File: Server/src/main/content/global/skill/production/cooking/recipe/topping/impl/EggAndTomato.java
package content.global.skill.production.cooking.recipe.topping.impl;

import content.global.skill.production.cooking.recipe.Recipe;
import core.game.interaction.NodeUsageEvent;
import core.game.node.entity.player.Player;
import core.game.node.entity.skill.Skills;
import core.game.node.item.Item;

/**
 * The Egg and tomato.
 */
public class EggAndTomato extends Recipe {

    private static final Item EGG_AND_TOMATO = new Item(7064);

    private static final Item SCRAMBLED_EGG = new Item(7078);

    private static final Item TOMATO = new Item(1982);

    @Override
    public void mix(final Player player, final NodeUsageEvent event) {
        if (player.getSkills().getLevel(Skills.COOKING) < 23) {
            player.getDialogueInterpreter().sendDialogue("You need a Cooking level of at least " + 23 + " in order to do this.");
            return;
        }
        super.mix(player, event);
        player.getSkills().addExperience(Skills.COOKING, 50, true);
    }

    @Override
    public Item getBase() {
        return SCRAMBLED_EGG;
    }

    @Override
    public Item getProduct() {
        return EGG_AND_TOMATO;
    }

    @Override
    public Item[] getIngredients() {
        return new Item[]{TOMATO};
    }

    @Override
    public Item[] getParts() {
        return new Item[]{};
    }

    @Override
    public String getMixMessage(NodeUsageEvent event) {
        return null;
    }

    @Override
    public boolean isSingular() {
        return true;
    }

}

File: Emulator/src/main/content/global/skill/cooking/recipe/topping/impl/EggAndTomato.kt
package content.global.skill.cooking.recipe.topping.impl

import content.global.skill.cooking.recipe.Recipe
import org.rs.consts.Items
import core.api.getStatLevel
import core.api.rewardXP
import core.api.sendDialogue
import core.game.interaction.NodeUsageEvent
import core.game.node.entity.player.Player
import core.game.node.entity.skill.Skills
import core.game.node.item.Item

/**
 * Egg and tomato.
 */
class EggAndTomato : Recipe() {

    override fun mix(player: Player, event: NodeUsageEvent) {
        if (getStatLevel(player, Skills.COOKING) < 23) {
            sendDialogue(player, "You need a Cooking level of at least " + 23 + " in order to do this.")
            return
        }
        super.mix(player, event)
        rewardXP(player, Skills.COOKING, 50.0)
    }

    override fun getBase(): Item {
        return SCRAMBLED_EGG
    }

    override fun getProduct(): Item {
        return EGG_AND_TOMATO
    }

    override fun getIngredients(): Array<Item> {
        return arrayOf(TOMATO)
    }

    override fun getParts(): Array<Item> {
        return arrayOf()
    }

    override fun getMixMessage(event: NodeUsageEvent): String? {
        return null
    }

    override fun isSingular(): Boolean {
        return true
    }

    companion object {
        private val EGG_AND_TOMATO = Item(Items.EGG_AND_TOMATO_7064)
        private val SCRAMBLED_EGG = Item(Items.SCRAMBLED_EGG_7078)
        private val TOMATO = Item(Items.TOMATO_1982)
    }
}
